src/NuGet.Core/NuGet.Configuration/Settings/SettingsLoadingContext.cs

```diff
@@ -14,9 +14,7 @@ namespace NuGet.Configuration
     /// </summary>
     public sealed class SettingsLoadingContext : IDisposable
     {
-        private readonly Dictionary<string, SettingsFile> _cache = new Dictionary<string, SettingsFile>(PathUtility.GetStringComparerBasedOnOS());
-
-        private readonly object _lockObject = new object();
+        private readonly ConcurrentDictionary<string, Lazy<SettingsFile>> _cache = new ConcurrentDictionary<string, Lazy<SettingsFile>>(PathUtility.GetStringComparerBasedOnOS());
```

Comment:
the title suggests only the concurrent dictionary chnages, but the real change is the Lazy correct? 

Beyond that, the concurrentdictionary and the one with a writer lock, they're equivalent correct?

Reply:
> the title suggests only the concurrent dictionary chnages, but the real change is the Lazy correct?

Yes sorry, the pattern of any mechanism that uses a `ConcurrentDictionary<TKey, TValue>` where you only want `TValue` fetched once has to use `Lazy<T>` so I didn't explicitly call that out but meant to imply it.  The "real" change is to guarantee that the `SettingsLoadingContext` is thread-safe and does not cache exceptions for other settings file.


> Beyond that, the concurrentdictionary and the one with a writer lock, they're equivalent correct?

Apparently not because I've seen three failed builds where the current code is throwing the exception listed in https://github.com/NuGet/Home/issues/12208.  If I'm going to have to `lock()` all access to the `Dictionary`, I figured I might as well use this pattern now instead.

Reply:
I see. I don't care enoguh to try to figure out now :D I think the change as is makes sense.